shard-operator-parent/shard-operator-core/src/test/java/com/redhat/service/smartevents/shard/operator/core/providers/TestResourceLoader.java

```diff
@@ -0,0 +1,25 @@
+package com.redhat.service.smartevents.shard.operator.core.providers;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.utils.Serialization;
+
+public class TestResourceLoader extends TemplateProviderImpl {
+
+    private static final String TEMPLATES_DIR = "/templates";
+    private static final String TEST_DEPLOYMENT_PATH = TEMPLATES_DIR + "/test_deployment.yaml";
+
+    public Deployment loadTestDeployment() {
+        return loadYaml(Deployment.class, TEST_DEPLOYMENT_PATH);
+    }
+
+    private static <T> T loadYaml(Class<T> clazz, String yaml) {
+        try (InputStream is = TemplateProviderImpl.class.getResourceAsStream(yaml)) {
+            return Serialization.unmarshal(is, clazz);
+        } catch (IOException ex) {
+            throw new IllegalStateException("Cannot find yaml on classpath: " + yaml);
+        }
+    }
```

Comment:
Now that `TestResourceLoader` extends `TemplateProviderImpl` it should be possible to make [`loadYaml`](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/shard-operator-parent/shard-operator-core/src/main/java/com/redhat/service/smartevents/shard/operator/core/providers/TemplateProviderImpl.java#L136) code protected and remove this code.
Can you try please?